toolchain/parse/extract.cpp

```diff
@@ -210,49 +210,35 @@ struct Extractable<std::optional<T>> {
   }
 };
 
-// Extract a `tuple<T...>` from a list of child nodes by extracting each `T` in
-// reverse order.
-template <typename... T>
-struct Extractable<std::tuple<T...>> {
-  template <std::size_t... Index>
-  static auto ExtractImpl(const Tree* tree, Tree::SiblingIterator& it,
-                          Tree::SiblingIterator end, ErrorBuilder* trace,
-                          std::index_sequence<Index...>)
-      -> std::optional<std::tuple<T...>> {
-    std::tuple<std::optional<T>...> fields;
-    if (trace) {
-      *trace << sizeof...(T) << "-tuple: begin\n";
-    }
-
-    // Use a fold over the `=` operator to parse fields from right to left.
-    [[maybe_unused]] int unused;
-    bool ok = true;
-    static_cast<void>(
-        ((ok && (ok = (std::get<Index>(fields) =
-                           Extractable<T>::Extract(tree, it, end, trace))
-                          .has_value()),
-          unused) = ... = 0));
-
-    if (!ok) {
-      if (trace) {
-        *trace << sizeof...(T) << "-tuple: error\n";
-      }
-      return std::nullopt;
-    }
-
+template <typename T, typename... U, std::size_t... Index>
+static auto ExtractTupleLikeType(const Tree* tree, Tree::SiblingIterator& it,
+                                 Tree::SiblingIterator end, ErrorBuilder* trace,
+                                 std::index_sequence<Index...>,
+                                 std::tuple<U...>*) -> std::optional<T> {
```

Comment:
```suggestion
                                 std::index_sequence<Index...> /*indices*/,
                                 std::tuple<U...>* /*type*/) -> std::optional<T> {
```

Suggesting names for [clang-tidy](https://clang.llvm.org/extra/clang-tidy/checks/readability/named-parameter.html)